models/oneinch/ethereum/unoswap/oneinch_unoswap_v5_ethereum_trades.sql

```diff
@@ -0,0 +1,261 @@
+{{  config(
+        schema='oneinch_unoswap_v5_ethereum',
+        alias='trades',
+        partition_by = ['block_date'],
+        on_schema_change='sync_all_columns',
+        file_format ='delta',
+        materialized='incremental',
+        incremental_strategy='merge',
+        unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address']
+    )
+}}
+
+{% set project_start_date = '2022-11-07' %} --for testing, use small subset of data
+{% set generic_null_address = '0x0000000000000000000000000000000000000000' %} --according to etherscan label
+{% set burn_address = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee' %} --according to etherscan label
+{% set blockchain = 'ethereum' %}
+{% set blockchain_symbol = 'ETH' %}
+
+WITH unoswap AS
+(
+    SELECT
+        call_block_number
+        ,output_returnAmount
+        ,amount
+        ,srcToken
+        ,pools
+        ,call_tx_hash
+        ,call_trace_address
+        ,call_block_time
+        ,contract_address
+    FROM
+        {{ source('oneinch_ethereum', 'AggregationRouterV5_call_unoswap') }}
+    WHERE
+        call_success
+        /******************************************************************************************************************
+            - two tx's don't fit into the join on line 120:
+                AND COALESCE(array_size(unoswap.call_trace_address), 0) + 2 = COALESCE(array_size(traces.trace_address), 0)
+            - the '+ 2' should apparently be '+ 3' for these tx's to correctly join to traces
+            - on v1 engine, the tx's were forced to amount_usd = 0 via update statement, as full refresh is less common there
+        ********************************************************************************************************************/
+        AND call_tx_hash not in (
```

Comment:
two edge case tx's were applying the wrong token bought address, therefore value, which blew up `amount_usd`. comments explain the join condition which brings about the issue. since it's two tx's, we'll filter out for now and look to see how to enhance the join. a few attempts to fix in dev were blowing up other tx's incorrectly.